File: expat.lua
local lxp = require "lxp"

local state = 0

function start_element(parser, name)
    if name == "ProductUrl" then
        state = 1
    end
end

function char_data(parser, data)
    if state == 1 then
        print (data)
        state = 0
    end
end

local p = lxp.new({StartElement = start_element, CharacterData = char_data})
local f = io.open(arg[1], 'r')

local done = false
while not done do
    local l = f:read(65536)
    done = l:len() ~= 65536
    p:parse(l)
end
p:parse()
p:close()
f:close()

File: expat.py
import sys
import xml.parsers.expat

state = 0

def start_element(name, attrs):
    global state
    if name == 'ProductUrl':
        state = 1


def char_data(data):
    global state
    if state == 1:
        print (data)
        state = 0


def get_producturls(f):
    p = xml.parsers.expat.ParserCreate()
    p.StartElementHandler = start_element
    p.CharacterDataHandler = char_data
    p.ParseFile(f)


def entry_point(argv):
    if len(argv) != 2:
        print ("Usage: %s <file>" % argv[0])
        return 1

    f = open(argv[1], 'rb')
    get_producturls(f)
    f.close()
    return 0


if __name__ == '__main__':
    entry_point(sys.argv)
